test/lib/cli/upgrade_command_test.rb

```diff
@@ -228,6 +228,7 @@ def foreman_maintain_update_unavailable
       end
 
       it 'with --phase it runs only a specific phase of the upgrade' do
+        foreman_maintain_update_unavailable
```

Comment:
This change means that even if we're running with `--phase` it tries to update itself?
IMHO that sounds wrong

Reply:
I don't know what it's doing.

Reply:
My understanding is that one can tell FM to only run a portion of an upgrade and your change affects the selection of executed commands.

I am not 100% sure this is bad or not, but something we should investigate (but that's IMHO unrelated to the other comments in this PR)

Reply:
If the phase is part of the upgrade procedure, I would expect self-upgrade to still attempt to happen.

Reply:
So, if the phase test would actually assert the *output*, we'd have seen the following:
```
+"Checking for new version of rubygem-foreman_maintain...
+undefined method `module_exists?' for #<ForemanMaintain::PackageManager::Apt:0xXXXXXX>
+
+      if package_manager.module_exists?(maintenance_module) &&
+                        ^^^^^^^^^^^^^^^
+"
```

Which is correct -- you only implemented that for DNF, as that's only available on DNF, but you call it on all OSes.

Reply:
This fixes the issue for me:

```patch
diff --git lib/foreman_maintain.rb lib/foreman_maintain.rb
index 584b65f..e5bbb57 100644
--- lib/foreman_maintain.rb
+++ lib/foreman_maintain.rb
@@ -192,6 +192,8 @@ module ForemanMaintain
     end
 
     def enable_maintenance_module
+      return unless el?
+
       maintenance_module = 'satellite-maintenance:el8'
       package_manager = ForemanMaintain.package_manager
 
diff --git test/lib/cli/upgrade_command_test.rb test/lib/cli/upgrade_command_test.rb
index 16b71b1..5806e27 100644
--- test/lib/cli/upgrade_command_test.rb
+++ test/lib/cli/upgrade_command_test.rb
@@ -228,9 +228,11 @@ module ForemanMaintain
       end
 
       it 'with --phase it runs only a specific phase of the upgrade' do
-        foreman_maintain_update_unavailable
         UpgradeRunner.any_instance.expects(:run_phase).with(:pre_migrations)
-        run_cmd(['--phase=pre_migrations', '--target-version=1.15'])
+        assert_cmd(<<-OUTPUT.strip_heredoc, ['--phase=pre_migrations', '--target-version=1.15'])
+          Checking for new version of rubygem-foreman_maintain...
+          Nothing to update, can't find new version of rubygem-foreman_maintain.
+        OUTPUT
       end
 
       it 'raises an exception for invalid phase' do
```

Reply:
We could also make the `PackageManager::Base` have a `modularity_supported?` returning `false` and `PackageManager::Dnf` with `true` and check on that. I don't mind either way, but we need a fix to not break Debian upgrades.

Reply:
(Also, I think our tests today depend too much on the OS they are running on, as we don't have any `on_supported_os` helper like in Puppet)

Reply:
That commit you pushed looks good to me, thanks for figuring out that output test as well! So if you like it, approve and merge it.